File: src/main/java/be4rjp/sclat/manager/WeaponClassMgr.java

package be4rjp.sclat.manager;

import be4rjp.sclat.Sclat;
import be4rjp.sclat.api.config.WeaponConfig;
import be4rjp.sclat.api.player.PlayerData;
import be4rjp.sclat.data.DataMgr;
import be4rjp.sclat.data.WeaponClass;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static be4rjp.sclat.Sclat.conf;

/**
 *
 * @author Be4rJP
 */
public class WeaponClassMgr {
	private static final Logger logger = LoggerFactory.getLogger(WeaponClassMgr.class);

	public synchronized static void WeaponClassSetup() {
		ConfigurationSection weaponClassSection = conf.getClassConfig().getConfigurationSection("WeaponClass");
		if(weaponClassSection == null) {
			logger.warn("weaponClassSection is null");
			return;
		}

		for (String classname : weaponClassSection.getKeys(false)) {
			ConfigurationSection weaponSection = weaponClassSection.getConfigurationSection(classname);
			if(weaponSection == null) {
				logger.warn("weaponSection of {} is null", classname);
				continue;
			}

			WeaponClass wc = WeaponConfig.parseSection(classname, weaponSection);
			DataMgr.setWeaponClass(classname, wc);
		}
	}

	public static void setWeaponClass(Player player) {
		player.getInventory().clear();
		PlayerData data = DataMgr.getPlayerData(player);
		ItemStack main = data.getWeaponClass().getMainWeapon().getWeaponIteamStack().clone();
		if (data.getMainItemGlow()) {
			Sclat.glow.enchantGlow(main);
			main.addEnchantment(Sclat.glow, 1);
		}
		player.getInventory().setItem(0, main);
		if (data.getWeaponClass().getMainWeapon().getIsManeuver())
			player.getInventory().setItem(40, data.getWeaponClass().getMainWeapon().getWeaponIteamStack().clone());
		ItemStack is = SubWeaponMgr.getSubWeapon(player);
		player.getInventory().setItem(2, is);
		ItemStack co = new ItemStack(Material.BOOK);
		ItemMeta meta = co.getItemMeta();
		meta.setDisplayName("スーパージャンプ");
		co.setItemMeta(meta);
		player.getInventory().setItem(6, co);
		if (!data.getIsSquid())
			player.getEquipment().setHelmet(DataMgr.getPlayerData(player).getTeam().getTeamColor().getBougu());

		if (data.getSPGauge() == 100)
			SPWeaponMgr.setSPWeapon(player);

		if (conf.getConfig().getString("WorkMode").equals("Trial") && !Sclat.tutorial) {
			ItemStack join = new ItemStack(Material.CHEST);
			ItemMeta joinmeta = join.getItemMeta();
			joinmeta.setDisplayName(ChatColor.GOLD + "右クリックでメインメニューを開く");
			join.setItemMeta(joinmeta);
			player.getInventory().setItem(7, join);
		}
	}

}

File: src/main/kotlin/be4rjp/sclat/manager/WeaponClassMgr.kt
package be4rjp.sclat.manager

import be4rjp.sclat.Sclat
import be4rjp.sclat.data.DataMgr.getPlayerData
import be4rjp.sclat.data.DataMgr.getWeapon
import be4rjp.sclat.data.DataMgr.setWeaponClass
import be4rjp.sclat.data.WeaponClass
import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

/**
 *
 * @author Be4rJP
 */
object WeaponClassMgr {
    @Synchronized
    fun weaponClassSetup() {
        for (classname in Sclat.conf!!
            .classConfig!!
            .getConfigurationSection("WeaponClass")!!
            .getKeys(false)) {
            val weaponName =
                Sclat.conf!!
                    .classConfig!!
                    .getString("WeaponClass." + classname + ".MainWeaponName")
            val subWeaponName =
                Sclat.conf!!
                    .classConfig!!
                    .getString("WeaponClass." + classname + ".SubWeaponName")
            val spWeaponName =
                Sclat.conf!!
                    .classConfig!!
                    .getString("WeaponClass." + classname + ".SPWeaponName")
            val wc = WeaponClass(classname)
            wc.mainWeapon = (getWeapon(weaponName))
            wc.subWeaponName = subWeaponName
            wc.sPWeaponName = spWeaponName

            setWeaponClass(classname, wc)
        }
    }

    @JvmStatic
    fun setWeaponClass(player: Player) {
        player.inventory.clear()
        val data = getPlayerData(player)
        val main =
            data!!
                .weaponClass!!
                .mainWeapon!!
                .weaponIteamStack!!
                .clone()
        if (data.mainItemGlow) {
            Sclat.glow!!.enchantGlow(main)
            main.addEnchantment(Sclat.glow!!, 1)
        }
        player.inventory.setItem(0, main)
        if (data.weaponClass!!.mainWeapon!!.isManeuver) {
            player
                .inventory
                .setItem(
                    40,
                    data.weaponClass!!
                        .mainWeapon!!
                        .weaponIteamStack!!
                        .clone(),
                )
        }
        val `is` = SubWeaponMgr.getSubWeapon(player)
        player.inventory.setItem(2, `is`)
        val co = ItemStack(Material.BOOK)
        val meta = co.itemMeta
        meta!!.setDisplayName("スーパージャンプ")
        co.itemMeta = meta
        player.inventory.setItem(6, co)
        if (!data.isSquid) player.equipment!!.helmet = getPlayerData(player)!!.team!!.teamColor!!.bougu

        if (data.sPGauge == 100) SPWeaponMgr.setSPWeapon(player)

        if (Sclat.conf!!
                .config!!
                .getString("WorkMode") == "Trial" &&
            !Sclat.tutorial
        ) {
            val join = ItemStack(Material.CHEST)
            val joinmeta = join.itemMeta
            joinmeta!!.setDisplayName(ChatColor.GOLD.toString() + "右クリックでメインメニューを開く")
            join.itemMeta = joinmeta
            player.inventory.setItem(7, join)
        }
    }
}
